crates/ra_hir/src/semantics.rs

```diff
@@ -80,6 +82,13 @@ impl PathResolution {
     }
 }
 
+pub enum SelfKind {
+    Shared,
+    Mutable,
+    Consuming,
```

Comment:
Should I just use existing rust terminology like `Moved` or `Owned`? Same for the new `HighlightModifier`.